libcontainer/process_linux.go

```diff
@@ -420,7 +420,24 @@ func (p *initProcess) terminate() error {
 	if _, werr := p.wait(); err == nil {
 		err = werr
 	}
-	return err
+	return ignoreTerminateErrors(err)
+}
+
+// ignoreTerminateErrors returns nil if the given err matches an error known
+// to indicate that the terminate occurred successfully or err was nil, otherwise
+// err is returned unaltered.
+func ignoreTerminateErrors(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	// TODO(steve): Update these to none string checks if the runtime exports them.
+	switch err.Error() {
+	case "os: process already finished", "exec: Wait was already called":
```

Comment:
If this was an public method I would totally agree, but as this is private and used solely for error recovery this seemed good.

However you are indeed correct that even though its private additional consumers of terminate could still be added where this behaviour is not desired, so I've moved the check to the consumer instead.